fetch.js

```diff
@@ -177,6 +194,14 @@
         this._bodyText = body.toString()
       } else if (!body) {
         this._bodyText = ''
+      } else if (body.getReader) {
+        this.body = body
+        Object.defineProperty(this, 'bodyUsed', {
```

Comment:
I wasn't a big fan of this either, I even asked the community if it was okey to do this: https://github.com/whatwg/fetch/issues/396

But it wasn't... We should maybe implement a new TypeError message that checks the locked property. 
It's a bit more complicated to set the bodyUsed property since it needs set it upon the first getReader().read call and spying on this is not that easy...